BasePreparedModel.cpp

```diff
@@ -424,12 +426,145 @@ Return<V1_3::ErrorStatus> BasePreparedModel::execute_1_3(
     const V1_3::OptionalTimeoutDuration& loopTimeoutDuration,
     const V1_3::OptionalTimeoutDuration& duration, executeFenced_cb cb) {
     ALOGV("Entering %s", __func__);
-    // TODO: Add support
+
+    time_point driverStart, driverEnd, deviceStart, deviceEnd, driverAfterFence;
+    if (measure == MeasureTiming::YES) driverStart = now();
+
+    if (!validateRequest(request1_3, mModelInfo->getModel(), /*allowUnspecifiedOutput=*/false)) {
+        cb(V1_3::ErrorStatus::INVALID_ARGUMENT, hidl_handle(nullptr), nullptr);
+        return Void();
+    }
+
+    auto errorStatus = mModelInfo->setRunTimePoolInfosFromHidlMemories(request1_3.pools);
+    if (errorStatus != V1_3::ErrorStatus::NONE) {
+        cb(errorStatus, hidl_handle(nullptr), nullptr);
+        return Void();
+    }
+
+    // rest of the interfaces are based on 1.0 request
+    auto request = convertToV1_0(request1_3);
+
+    // Wait for the dependent events to signal
+    for (const auto& fenceHandle : waitFor) {
+        if (!fenceHandle.getNativeHandle()) {
+            cb(V1_3::ErrorStatus::INVALID_ARGUMENT, hidl_handle(nullptr), nullptr);
+            return Void();
+        }
+        int syncFenceFd = fenceHandle.getNativeHandle()->data[0];
+        if (syncWait(syncFenceFd, -1) != FenceState::SIGNALED) {
+            ALOGV("%s syncWait failed", __func__);
+            cb(V1_3::ErrorStatus::GENERAL_FAILURE, hidl_handle(nullptr), nullptr);
+            return Void();
+        }
+    }
+    if (measure == MeasureTiming::YES) driverAfterFence = now();
+
+    for (size_t i = 0; i < request.inputs.size(); i++) {
+        auto inIndex = mModelInfo->getModelInputIndex(i);
+        auto srcBlob = mModelInfo->getBlobFromMemoryPoolIn(request, i);
+
+        const std::string& inputNodeName = mNgc->getNodeName(inIndex);
+        if (inputNodeName == "") {
+            ALOGD("Ignorning input at index(%d), since it is invalid", inIndex);
+            continue;
+        }
+        ALOGD("Input index: %d layername : %s", inIndex, inputNodeName.c_str());
+        auto destBlob = mPlugin->getBlob(inputNodeName);
+        uint8_t* dest = destBlob->buffer().as<uint8_t*>();
+        uint8_t* src = srcBlob->buffer().as<uint8_t*>();
+        std::memcpy(dest, src, srcBlob->byteSize());
+    }
+    ALOGD("%s Run", __func__);
+
+    if (measure == MeasureTiming::YES) deviceStart = now();
+    try {
+        mPlugin->infer();
+    } catch (const std::exception& ex) {
+        ALOGE("%s Exception !!! %s", __func__, ex.what());
+        cb(V1_3::ErrorStatus::GENERAL_FAILURE, hidl_handle(nullptr), nullptr);
+        return Void();
+    }
+    if (measure == MeasureTiming::YES) deviceEnd = now();
+
+    for (size_t i = 0; i < request.outputs.size(); i++) {
+        auto outIndex = mModelInfo->getModelOutputIndex(i);
+        ALOGI("OutputIndex: %d", outIndex);
+        const std::string& outputNodeName = mNgc->getNodeName(outIndex);
```

Comment:
mNgc isn't a very meaningful var name. I know its not added by you, Can we change this to mNgraphNetCreator?